source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -57,6 +57,8 @@ void GrpcClientImpl::check(RequestCallbacks& callbacks, Event::Dispatcher& dispa
     }
   }
 
+  options.setParentContext(Http::AsyncClient::ParentContext{&stream_info});
```

Comment:
It might be good one day to be able to have async clients automatically detect this information, maybe by introducing a StreamAsyncClient that can pick up on this context. In any case, I think what you have is fine for now.